extensions/amp-story/1.0/amp-story-embedded-component.js

```diff
@@ -536,28 +547,57 @@ export class AmpStoryEmbeddedComponent {
           const targetRect = target./*OK*/getBoundingClientRect();
           const pageRect = this.componentPage_./*OK*/getBoundingClientRect();
 
-          const centeredTop = pageRect.height / 2 - targetRect.height / 2;
-          const centeredLeft = pageRect.width / 2 - targetRect.width / 2;
+          this.scaleFactor_ =
+            parseFloat(getStyle(target,'transform').split('scale(')[1]);
+          const realWidth = targetRect.width / this.scaleFactor_;
+          const leftOffset = (realWidth - targetRect.width) / 2;
+          state.translateX = (targetRect.left - leftOffset - pageRect.left);
 
-          // Only account for offset from target to page borders. Since in
-          // desktop mode page is not at the borders of viewport.
-          const leftOffset = targetRect.left - pageRect.left;
-          const topOffset = targetRect.top - pageRect.top;
+          const realHeight = targetRect.height / this.scaleFactor_;
+          const topOffset = (realHeight - targetRect.height) / 2;
+          const verticalOffset = targetRect.top - topOffset - pageRect.top;
+          const centeredTop = pageRect.height / 2 - realHeight / 2;
 
-          state.translateY = centeredTop - topOffset;
-          state.translateX = leftOffset - centeredLeft;
+          state.translateY = centeredTop - verticalOffset;
         },
         /** mutate */
         () => {
           target.classList.add('i-amphtml-animate-expand-in');
           setImportantStyles(dev().assertElement(target),
               {
-                transform: `translate3d(${state.translateX}px,
-                    ${state.translateY}px, 0)`,
+                transform: `translate3d(${-state.translateX}px,
+                  ${state.translateY}px, 0) scale(1)`,
               });
         });
   }
 
+  /**
+   * Prepares expandable component for its expanded animation.
+   * It resizes it to its full-screen size, and scales it down to match size set
+   * by publisher, adding negative margins so that content around stays put.
+   * @param {!Element} page
+   * @param {!Element} element
+   */
+  static prepareForAnimation(page, element) {
+    element.whenBuilt().then(() => {
+      const pageRect = page./*OK*/getBoundingClientRect();
+      const elRect = element./*OK*/getBoundingClientRect();
+
+      const scaleFactor = elRect.width / pageRect.width;
```

Comment:
This might need to change; this assumes that the component is wider than it is tall, right?  I think we'd want to scale based on width assuming a wide component, but scale based on height assuming a tall component.

...I think?